### PR TITLE
Restore auth form contextual help.

### PR DESCRIPTION
**Pull request by n1k0**
We recently mistakenly removed contextual help texts in AuthForm. This patch reintroduces them and cleans code up a bit.

### CODE REVIEW

File: src/components/AuthForm.js [hunk 1]
```diff
@@ -346,12 +348,25 @@ export default class AuthForm extends Component {
...
+      singleAuthMethod
+    );
     return (
       <div className="panel panel-default">
         <div className="panel-body">
           <BaseForm
-            schema={extendSchemaWithHistory(schema, history, authMethods, singleServer)}
-            uiSchema={extendUiSchemaWithHistory(uiSchema, history, clearHistory, singleServer, singleAuthMethod)}
+            schema={finalSchema}
+            uiSchema={finalUiSchema}
```
Comment (glasserc) [inline src/components/AuthForm.js]: This change is just a refactor, right? Or is this related to the fix?

File: src/components/AuthForm.js [hunk 2]
```diff
@@ -313,8 +314,9 @@ export default class AuthForm extends Component {
   onChange = ({formData}: {formData: Object}) => {
     const {authType} = formData;
     const {schema, uiSchema} = authSchemas[authType];
-    const specificFormData = /fxa|anonymous/.test(authType) ? omit(formData, ["credentials"])
-                                                            : {credentials: {}, ...formData};
+    const specificFormData = ["anonymous", "fxa", "portier"].includes(authType)
+      ? omit(formData, ["credentials"])
+      : {credentials: {}, ...formData};
```
Comment (glasserc) [inline src/components/AuthForm.js]: Is this related to the fix? It looks like it's an unrelated fix for Portier.

Comment (n1k0) [inline src/components/AuthForm.js]: Yeah we forgot to clean credentials when handling Portier auth form data in #395.

File: src/components/AuthForm.js (hunk 1 above)
Comment (n1k0) [inline src/components/AuthForm.js]: Just code formatting to improve legibility.
